driver/csiplugin/gpfs_util.go

```diff
@@ -99,6 +102,19 @@ type scaleVolSnapId struct {
 	SnapId    string
 } //nolint
 
+func IsValidCompressionAlgorithm(input string) bool {
+	switch strings.ToLower(input) {
+	case
+		"z",
+		"lz4",
+		"zfast",
+		"alphae",
+		"alphah":
```

Comment:
are we also ensuing that these  compression algorithm exists on the scale cluster before allowing user to create PVC?

Reply:
Yes, we are looking at the pre-reqs that are needed for this. There is the gpfs.compression rpm that installs various glue code functionality which should contain all the relevant libraries.

We did test this in the CNSA cluster and we were able to compress files with all algorithms with no additional rpms needed. We can run some additional tests to see if there is a pre-check we can run as part of CreateVolume to make sure everything is installed.